table-api/src/main/java/io/deephaven/api/filter/FilterIsNotNull.java

```diff
@@ -3,33 +3,42 @@
  */
 package io.deephaven.api.filter;
 
-import io.deephaven.api.ColumnName;
 import io.deephaven.annotations.SimpleStyle;
+import io.deephaven.api.expression.Expression;
 import org.immutables.value.Value.Immutable;
 import org.immutables.value.Value.Parameter;
 
 /**
- * Evaluates to {@code true} when the given {@link #column() column} value is not {@code null}.
+ * Evaluates to {@code true} when the given {@link #expression() expression} evaluates to not {@code null}.
  */
 @Immutable
 @SimpleStyle
 public abstract class FilterIsNotNull extends FilterBase {
 
-    public static FilterIsNotNull of(ColumnName column) {
-        return ImmutableFilterIsNotNull.of(column);
+    public static FilterIsNotNull of(Expression expression) {
+        return ImmutableFilterIsNotNull.of(expression);
```

Comment:
Isn't this going to throw, since we only support checking non-null of a column name (`Only supports null checking a reference to a column`)?

Reply:
From a logical perspective, we should be able to express a nullness check against an `Expression`. The limitations of the gRPC layer is called out in the PR description - I definitely think we should work on closing the gap, but it wasn't top-of-priority given that SQL is server-side (script session) only ATM.

Reply:
#3609